README.md

```diff
@@ -1,47 +1,96 @@
+Causal testing is a causal inference-driven framework for functional black-box testing. This framework utilises
+graphical causal inference (CI) techniques for the specification and functional testing of software from a black-box
+perspective. In this framework, we use causal directed acyclic graphs (DAGs) to express the anticipated cause-effect
+relationships amongst the inputs and outputs of the system-under-test and the supporting mathematical framework to
+design statistical procedures capable of making causal inferences. Each causal test case focuses on the causal effect of
+an intervention made to the system-under test. That is, a prescribed change to the input configuration of the
+system-under-test that is expected to cause a change to some output(s).
 
 ![Causal Testing Workflow](images/workflow.png)
 
 The causal testing framework has three core components:
 
-1. [Causal specification](causal_testing/specification/README.md): Before we can test software, we need to obtain an understanding of how it should behave in a particular use-case scenario. In addition, to apply graphical CI techniques for testing, we need a causal DAG which depicts causal relationships amongst inputs and outputs. To collect this information, users must create a _causal specification_. This comprises a set of scenarios which place constraints over input variables that capture the use-case of interest, a causal DAG corresponding to this scenario, and a series of high-level functional requirements that the user wishes to test. In causal testing, these requirements should describe how the model should respond to interventions (changes made to the input configuration).
-
-2. [Causal tests](causal_testing/testing/README.md): With a causal specification in hand, we can now go about designing a series of test cases that interrogate the causal relationships of interest in the scenario-under-test. Informally, a causal test case is a triple (M, X, Delta, Y), where M is the modelling scenario, X is an input configuration, Delta is an intervention which should be applied to X, and Y is the expected _causal effect_ of that intervention on some output of interest. Therefore, a causal test case states the expected causal effect (Y) of a particular intervention (Delta) made to an input configuration (X). For each scenario, the user should create a suite of causal tests. Once a causal test case has been defined, it is executed as follows:
-    1. Using the causal DAG, identify an estimand for the effect of the intervention on the output of interest. That is, a statistical procedure capable of estimating the causal effect of the intervention on the output.
+1. [Causal specification](causal_testing/specification/README.md): Before we can test software, we need to obtain an
```

Comment:
Should the target of this link be in docs? It's, on reflection, a little odd having md files in package...